src/core_seaice/shared/mpas_seaice_column.F

```diff
@@ -1608,11 +1600,13 @@ subroutine column_vertical_thermodynamics(domain, clock)
                dayOfYear, &
                abortFlag, &
                abortMessage)
-          call colpkg_get_warnings(warnings)
-          call column_write_warnings(warnings, abortFlag)
-
+          call column_write_warnings(abortFlag)
           ! code abort
-          if (abortFlag) exit
+          if (abortFlag) then
+             call mpas_log_write("column_vertical_thermodynamics: "//trim(abortMessage) , messageType=MPAS_LOG_ERR)
```

Comment:
If this has been moved into the iCell loop, does the call to this after the loop not need to be removed?

Reply:
A call to `MPAS_LOG_CRIT`-type `mpas_log_write` should abort the run, if the abort flag is set inside colpkg calls. This makes the abort happen earlier: as soon as an error is detected.

There is a call to `mpas_dmpar_max_int` inside `call seaice_check_critical_error`, which I had to remove, because calls to MPI-related functions (MPI_Allreduce here) from inside OMP-threaded regions should be avoided.

If any thread inside any MPI process, sets `abortFlag`, the run should be terminated. If `abortFlag` is not set, the two removed calls don't do much. So this change should preserve previous behavior.

Reply:
So, these changes then remove the ability to perform a full abort stream write (seaice_check_critical_error), which is a significant degradation of the debugging capability of the model for coupled simulations. Isn't an alternative to set a flag inside the loop that an error occurred, let the loop finish and then outside the iCell loop abort the model with all the checks present?